packages/website/src/components/ASTViewerESTree.tsx

```diff
@@ -1,52 +1,32 @@
-import React, { useCallback } from 'react';
+import React, { useEffect, useState } from 'react';
 
 import ASTViewer from './ast/ASTViewer';
-import { isRecord } from './ast/utils';
-import type { ASTViewerBaseProps, SelectedRange } from './ast/types';
-import { TSESTree } from '@typescript-eslint/website-eslint';
+import type { ASTViewerBaseProps, ASTViewerModelMap } from './ast/types';
+import type { TSESTree } from '@typescript-eslint/website-eslint';
+import { serialize } from './ast/serializer/serializer';
+import { createESTreeSerializer } from './ast/serializer/serializerESTree';
 
-function isESTreeNode(
-  value: unknown,
-): value is Record<string, unknown> & TSESTree.BaseNode {
-  return isRecord(value) && 'type' in value && 'loc' in value;
+export interface ASTESTreeViewerProps extends ASTViewerBaseProps {
+  readonly value: TSESTree.BaseNode | string;
 }
 
-export const propsToFilter = ['parent', 'comments', 'tokens'];
+export default function ASTViewerESTree({
+  value,
+  position,
+  onSelectNode,
+}: ASTESTreeViewerProps): JSX.Element {
+  const [model, setModel] = useState<string | ASTViewerModelMap>('');
 
-export default function ASTViewerESTree(
-  props: ASTViewerBaseProps,
-): JSX.Element {
-  const filterProps = useCallback(
-    (item: [string, unknown]): boolean =>
-      !propsToFilter.includes(item[0]) &&
-      !item[0].startsWith('_') &&
-      item[1] !== undefined,
-    [],
-  );
-
-  const getRange = useCallback(
-    (value: unknown): SelectedRange | undefined =>
-      isESTreeNode(value)
-        ? {
-            start: value.loc.start,
-            end: value.loc.end,
-          }
-        : undefined,
-    [],
-  );
-
-  const getNodeName = useCallback(
-    (value: unknown): string | undefined =>
-      isESTreeNode(value) ? String(value.type) : undefined,
-    [],
-  );
+  useEffect(() => {
+    if (typeof value === 'string') {
+      setModel(value);
+    } else {
+      const scopeSerializer = createESTreeSerializer();
+      setModel(serialize(value, scopeSerializer));
```

Comment:
was this copy pasta mis-naming?

```suggestion
      const astSerializer = createESTreeSerializer();
      setModel(serialize(value, astSerializer));
```

Reply:
yes